hldb/Lib/DirectConnector.swift

```diff
@@ -9,24 +9,460 @@
-    return nil
   }
   
+  public func executeUpdate(sql:String) -> Bool {
+    return self.executeUpdate(sql, withArgumentsInArray: [])
+  }
+  
+  public func executeUpdate(sql:String, withArgumentsInArray arrayArgs:[AnyObject]) -> Bool {
+    
+    print(sql)
+    print(arrayArgs)
+    
+    guard let _ = self.dbPointer else { return false }
+    
+    if (self.isExecutingStatement) {
+      NSLog("WARNING: DirectDB already executing statement")
+      return false;
+    }
+    
+    self.isExecutingStatement = true
+    
+    var rcs: Int32?
+    //var updateStatement: COpaquePointer = nil
+    var statement: COpaquePointer = nil
+
+    
+    /*if (_shouldCacheStatements) {
+     statement = [self cachedStatementForQuery:sql];
+     pStmt = statement ? [statement statement] : 0x00;
+     [statement reset];
+     }*/
+
+    
+    if (statement == nil) {
+      
+      rcs = sqlite3_prepare_v2(self.dbPointer!, sql, -1, &statement, nil);
+      
+      if (SQLITE_OK != rcs) {
+        if (self.loggingErrors) {
+          //Log error
+        }
+        let errorMessage = String.fromCString(sqlite3_errmsg(self.dbPointer!))!
+        print("Query could not be prepared! \(errorMessage)")
+        
+        sqlite3_finalize(statement)
+        self.isExecutingStatement = false
+        return false;
+      }
+    }
+    
+    
+    var idx = 0
+    let queryCount = Int(sqlite3_bind_parameter_count(statement))
+    if (arrayArgs.count != queryCount) {
+      NSLog("Error: the bind count (\(queryCount)) is not correct for the # of variables (\(arrayArgs.count)) (executeQuery)")
+      sqlite3_finalize(statement)
+      self.isExecutingStatement = false
+      return false
+    }
+    
+    for (idx, obj) in arrayArgs.enumerate() {
+      let res = self.bind(obj, atIndex:idx+1, inStatement:statement)
+      if (res == SQLITE_OK) {
+        print("bind was successful")
+      } else {
+        print("failed to bind: \(res)")
+      }
+    }
+    
+    /* Call sqlite3_step() to run the virtual machine. Since the SQL being
+     ** executed is not a SELECT statement, we assume no data will be returned.
+     */
+    
+    let rc = sqlite3_step(statement);
+    
+    if (SQLITE_DONE == rc) {
+      // all is well, let's return.
+    }
+    else if (SQLITE_ERROR == rc) {
+      if (self.loggingErrors) {
+        NSLog("Error calling sqlite3_step SQLITE_ERROR \(rc): \(sqlite3_errmsg(self.dbPointer!))");
+        NSLog("DB Query: %@", sql);
+      }
+    }
+    else if (SQLITE_MISUSE == rc) {
+      // uh oh.
+      if (self.loggingErrors) {
+        NSLog("Error calling sqlite3_step SQLITE_MISUSE \(rc): \(sqlite3_errmsg(self.dbPointer!))");
+        NSLog("DB Query: %@", sql);
+      }
+    }
+    else {
+      // wtf?
+      if (self.loggingErrors) {
+        NSLog("Error calling sqlite3_step UNKNOWN \(rc): \(sqlite3_errmsg(self.dbPointer!))");
+        NSLog("DB Query: %@", sql);
+      }
+    }
+    
+    if (rc == SQLITE_ROW) {
+      return false
```

Comment:
the sqlite state machine is not finalized whenever you break early out of this function; You can use the Swift-ism: `defer` to execute a block of code right before returning (no matter where you return)